### PR TITLE
[chore] Fix windowseventlogreceiver test flakiness

### LINKED ISSUE #34687: [receiver/windowseventlog] Flaky test TestReadWindowsEventLogger
### Component(s)

receiver/windowseventlog

### Describe the issue you're reporting

Flaky test: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10391822945/job/28775648414?pr=34656#step:6:571

```terminal
=== FAIL: . TestReadWindowsEventLogger (20.08s)
    receiver_windows_test.go:108: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/windowseventlogreceiver/receiver_windows_test.go:108
        	Error:      	Condition never satisfied
        	Test:       	TestReadWindowsEventLogger

=== FAIL: . TestReadWindowsEventLogger (re-run 1) (20.02s)
    receiver_windows_test.go:108: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/windowseventlogreceiver/receiver_windows_test.go:108
        	Error:      	Condition never satisfied
        	Test:       	TestReadWindowsEventLogger
```

### PR DESCRIPTION
**Pull request by pjanotti**
**Description:**
There are 3 main issues being addressed on this PR:

* There is a race between start and generating the events in the tests: the start require goroutines so a delay before generating the events will be advisable (not perfect but good enough per my tests)
* The tests assume that only the tests are generating events in that time window, that is impossible to guarantee on a windows box, so instead of checking for exactly one event they need to get new events and check the event provider instead
* The exclusion test checks if zero events were delivered by the receiver, it should check if one not filtered out was received, and this one should be generated after the event from the provider that should be excluded.

**Link to tracking Issue:**
Fixes #34687

**Testing:**
Multiple local runs, we should do a few runs on CI given the runner characteristics.

**Documentation:**
N/A

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]

File: receiver/windowseventlogreceiver/receiver_windows_test.go [hunk 1]
```diff
@@ -188,91 +168,68 @@ func TestReadWindowsEventLoggerRaw(t *testing.T) {
...
+			receiver, err := factory.CreateLogsReceiver(ctx, createSettings, cfg, sink)
+			require.NoError(t, err)
+
+			err = receiver.Start(ctx, componenttest.NewNopHost())
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, receiver.Shutdown(ctx))
+			}()
+			// Start launches nested goroutines, give them a chance to run before logging the test event(s).
+			time.Sleep(3 * time.Second)
```
Comment (atoulme) [inline receiver/windowseventlogreceiver/receiver_windows_test.go:215]: maybe consider a assert.EventuallyWithT to test this multiple times until it works

**Review by atoulme — APPROVED** [review]

File: receiver/windowseventlogreceiver/receiver_windows_test.go (hunk 1 above)
Comment (pjanotti) [reply receiver/windowseventlogreceiver/receiver_windows_test.go:215]: In principle, I really dislike sleeps on tests. I tried a few alternative before settling on this, however, I couldn't make it work reliably without a sleep.

**Review by dmitryax — APPROVED** [review]

**Comment by djaglowski** [comment]
@pjanotti, I'm seeing failures ([example](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10685441061/job/29625770119?pr=34720#step:6:231)) on #34720 which I think might be related to these changes. 

From what I can tell, we are failing a [`require.EventuallyWithT`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/bfdee420bce1e625c50c0f3629c6e26fe9dfab9b/receiver/windowseventlogreceiver/receiver_windows_test.go#L273) check, but then continuing with the assumption that it succeeded.

**Comment by pjanotti** [comment]
Thanks @djaglowski - I'll look into it.

**Pull request merged**